File: Python/underautomation/universal_robots/license/license_info.py
import typing
from underautomation.universal_robots.license.license_state import LicenseState
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.License import LicenseInfo as license_info

class LicenseInfo:
	def __init__(self, licenseIdentifier: str, licenseKey: str, _internal = 0):
		if(_internal == 0):
			self._instance = license_info(licenseIdentifier, licenseKey)
		else:
			self._instance = _internal
	def __repr__(self):
		return self._instance.ToString()
	@property
	def license_key(self) -> str:
		return self._instance.LicenseKey
	@property
	def product(self) -> str:
		return self._instance.Product
	@property
	def evaluation_days_left(self) -> typing.Any:
		return self._instance.EvaluationDaysLeft
	@property
	def evaluation_start_date(self) -> typing.Any:
		return self._instance.EvaluationStartDate
	@property
	def licensee(self) -> str:
		return self._instance.Licensee
	@property
	def trial_period_expiration_date(self) -> typing.Any:
		return self._instance.TrialPeriodExpirationDate
	@property
	def state(self) -> LicenseState:
		return LicenseState(self._instance.State)
	@property
	def product_release_date(self) -> typing.Any:
		return self._instance.ProductReleaseDate
	@property
	def maintenance_years(self) -> int:
		return self._instance.MaintenanceYears
	@property
	def license_issued_date(self) -> typing.Any:
		return self._instance.LicenseIssuedDate
	@property
	def maintenance_expiration_date(self) -> typing.Any:
		return self._instance.MaintenanceExpirationDate

File: License/LicenseInfo.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using System;

namespace License {
	/// <summary>
	/// Information about a license key
	/// </summary>
	public sealed class LicenseInfo {

		/// <summary>
		/// Create a new LicenseInfo instance to retrieve informations about a pair of identifier/key
		/// This class should not be used to register your product. Please use static function RegisterLicense to specify your license.
		/// </summary>
		/// <param name="licenseIdentifier">The name of your organization (null for trial version)</param>
		/// <param name="licenseKey">The key, associated to the identifier, supplied by UnderAutomation (null for trial version)</param>
		public LicenseInfo(string licenseIdentifier, string licenseKey)
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// The license key supplied by UnderAutomation (null for trial period)
		/// </summary>
		public string LicenseKey { get; }

		/// <summary>
		/// Commercial name of this .NET Software library
		/// </summary>
		public string Product { get; }

		/// <summary>
		/// Remaining days of the trial period. Null if the product is licensed. It could be negative if the trial period is ended since several days.
		/// </summary>
		public int? EvaluationDaysLeft { get; }

		/// <summary>
		/// The date the trial period starts. If the product is licensed, the date of the library first use.
		/// </summary>
		public DateTime EvaluationStartDate { get; }

		/// <summary>
		/// Name of your organisation
		/// </summary>
		public string Licensee { get; }

		/// <summary>
		/// The date the product will expire. Null if the product is licensed.
		/// </summary>
		public DateTime? TrialPeriodExpirationDate { get; }

		/// <summary>
		/// The current license state
		/// </summary>
		public LicenseState State { get; }

		/// <summary>
		/// The date this version of the product was released.
		/// </summary>
		public DateTime ProductReleaseDate { get; }

		/// <summary>
		/// Number of maintenance years included in your license
		/// </summary>
		public int MaintenanceYears { get; }

		/// <summary>
		/// The date you get the license
		/// </summary>
		public DateTime? LicenseIssuedDate { get; }

		/// <summary>
		/// The date your maintenance contract end and you no longer can use this license with newer versions.
		/// </summary>
		public DateTime? MaintenanceExpirationDate { get; }

		/// <summary>
		/// A human description of the license
		/// </summary>
		public override string ToString()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}
	}
}
